ldk/javascript/src/whisper/react/renderer-config.ts

```diff
@@ -0,0 +1,330 @@
+    type: Type,
+    oldProps: Props,
+    newProps: Props,
+    internalInstanceHandle: any,
+    keepChildren: boolean,
+    recyclableInstance: null | Instance,
+  ): Instance;
+
+  createContainerChildSet(container: Container): ChildSet;
+
+  appendChildToContainerChildSet(childSet: ChildSet, child: Instance | TextInstance): ChildSet;
+
+  finalizeContainerChildren(container: Container, newChildren: ChildSet): void;
+
+  replaceContainerChildren(container: Container, newChildren: ChildSet): void;
+
+  getOffscreenContainerType(): string;
+
+  getOffscreenContainerProps(mode: any, children: any): Props;
+
+  cloneHiddenInstance(
+    instance: Instance,
+    type: Type,
+    props: Props,
+    internalInstanceHandle: any,
+  ): Instance;
+
+  cloneHiddenTextInstance(
+    instance: Instance,
+    text: string,
+    internalInstanceHandle: any,
+  ): TextInstance;
+}
+
+export const config: CoreConfig & PersistenceConfig = {
+  afterActiveInstanceBlur: undefined,
+  akeOpaqueHydratingObject: undefined,
+  appendChildToContainerChildSet(childSet: ChildSet, child: Instance | TextInstance): ChildSet {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any -- just need to check type that doesn't exist in Instance
+    if ((child as any).type !== 'whisper') {
+      throw new Error('oh-whisper must be top-level element');
+    }
+    // In this function we are adding the Whisper properties to the ChildSet.
+    const whisper = (child as unknown) as NewWhisper;
+    childSet.label = whisper.label;
+    childSet.components = whisper.components;
+    return childSet;
+  },
+  appendInitialChild(parentInstance: Instance, child: Instance | TextInstance): void {
+    const result =
+      getHandlerByHelpsType(parentInstance.type).appendInitialChild?.(parentInstance, child) ||
+      false;
+    if (!result) {
+      console.warn(
+        `Cannot add item of type ${child.type} to parent of type ${parentInstance.type}`,
+      );
+    }
+  },
+  beforeActiveInstanceBlur: undefined,
+  cancelTimeout(id: HostConfigTimeoutHandle): void {
+    clearTimeout(id);
+  },
+  cloneHiddenInstance(
+    instance: Instance,
+    type: Type,
+    props: Props,
+    internalInstanceHandle: any,
+  ): Instance {
+    throw new Error('Not Implemented');
+  },
+  cloneHiddenTextInstance(
+    instance: Instance,
+    text: string,
+    internalInstanceHandle: any,
+  ): TextInstance {
+    throw new Error('Not Implemented');
+  },
+  cloneInstance(
+    instance: Instance,
+    updatePayload: Update | null,
+    type: Type,
+    oldProps: Props,
+    newProps: Props,
+    internalInstanceHandle: any,
+    keepChildren: boolean,
+    recyclableInstance: Instance | null,
+  ): Instance {
+    const value: any = {
+      ...instance,
+      ...newProps,
+    };
+    delete value.children;
+    if (!keepChildren && value.components) {
+      value.components = [];
+    }
+    getHandlerByTagType(type).assignTextChildren?.(instance, newProps);
```

Comment:
If we're cloning the instance, why are we assigning text children to the old instance and not to the "value"?

(I could be misunderstanding this function)

Reply:
Nah you got it right, this was an oversight that I didn't catch. Thanks for the catch :)